src/Utilities/Algorithm.hpp

```diff
@@ -0,0 +1,59 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <algorithm>
+#include <array>
+#include <cstddef>
+#include <numeric>
+
+#include "ErrorHandling/Assert.hpp"
+#include "Utilities/Gsl.hpp"
+
+/*!
+ * \ingroup UtilitiesGroup
+ * \brief Get the indices in the container `t` of the `K` elements that satisfy
+ * `op` after sorting with respect to `op`.
+ *
+ * Specifically, if `op` is `[](const lhs, const rhs) { return lhs < rhs; }`
+ * then the result will be the indices of the `K` smallest elements in `t`. If
+ * `op` is `[](const lhs, const rhs) { return lhs > rhs; }` then the result will
+ * be the indices of the `K` largest elements in `t`.
+ *
+ * \warning The indices are not sorted by `op` when returned. Use `std::sort` to
+ * sort them as follows:
```

Comment:
My suspicion is that most (all?) uses of `indices_of` will immediately sort the result. Why not provide this functionality directly, perhaps by adding a small function:

```
template <size_t K, typename T, typename Comparator>
auto sorted_indices_of(const T& t, Comparator op) noexcept -> std::array<size_t, K> {
  auto result = indices_of<K>(t, op);
  std::sort(result.begin(), ...);
  return result;
}
```